### PR TITLE
batch index and gathering data on tv shows / seasons to 300 at a time

### PR DESCRIPTION
**Pull request by l0ud0gg**
One of our clients was overwhelming the indexing of tv shows. This code will limit to only getting info on 300 seasons at a time (as we currently do with films https://github.com/shift72/kibble/blob/master/kibble/api/films.go#L73).  The function will be called over and over until all seasons are index https://github.com/shift72/kibble/blob/master/kibble/api/site.go#L113 i.e. if theres more than 300 seasons, it will batch the first lot, then there will still be seasons flagged as unresolved until all seasons are done.

### CODE REVIEW

File: kibble/api/tv.go [hunk 1]
```diff
@@ -74,6 +74,11 @@ func AppendAllTVShows(cfg *models.Config, site *models.Site, itemIndex models.It
 func AppendTVSeasons(cfg *models.Config, site *models.Site, slugs []string, itemIndex models.ItemIndex) error {
 
 	sort.Strings(slugs)
+
+	if len(slugs) > 300 {
+		slugs = slugs[:300]
+	}
+
```
Comment (sam-shift72) [inline kibble/api/tv.go:81]: This will slice the first 300, but do we need to do this in a loop to deal with the rest?

Comment (l0ud0gg) [reply kibble/api/tv.go:81]: its does the loop where its called https://github.com/shift72/kibble/blob/master/kibble/api/site.go#L113 which is the same as how it currently does the films

**Review by sam-shift72 — APPROVED** [review]
LGTM

**Review by sarge — APPROVED** [review]
LGTM

**Pull request merged**